server/ent/ent.js

```diff
@@ -90,7 +92,133 @@ function nmSearch(node) {
 }
 
 module.exports.strictSearch = strictSearch
-function strictSearch(node, nmInstances = [], globalNmList = []) {
+function strictSearch(node, fasta) {
+	let entResults = strictSearchHelper(node)
+	// Filter out nonmono pairs before returning 
+	let finalResults = {species:entResults.species,nm:[]}
+
+	const fastaData = parseFasta(fasta)
```

Comment:
Something is causing [this failing CI build](https://circleci.com/gh/hybsearch/hybsearch/532)—to me it seems like the `ent` checks on these files are what is causing problems.

@hawkrives, any ideas?

Reply:
"npm test -- -u" will update the test's expectations

Reply:
In the root project folder, fwiw

Reply:
Oh, so it's not actually failing, it's just that the new nonomono output is different from the old? 

Actually, on further inspection, it looks like an error is being thrown in `parseFastaIteratively`.  I think what's happening is I'm passing something that is not fasta to `parseFasta` function. I assumed all formats would be converted to fasta at that point in the pipeline. 

So we need a way to get the sequence of a given species that works regardless of how you input the data?

Reply:
>Oh, so it's not actually failing, it's just that the new nonomono output is different from the old?

Yeah, that's what Jest seems to be complaining about.

>Actually, on further inspection, it looks like an error is being thrown in `parseFastaIteratively`. I think what's happening is I'm passing something that is not fasta to `parseFasta` function. I assumed all formats would be converted to fasta at that point in the pipeline.

Yeah, that's what I observed in the error—the thing that you're passing to `parseFasta` doesn't respond to `split` (and it seems like it might be undefined, too)

Reply:
The test's only set up to provide the Newick tree to the test cases here. I'll poke and try to add the FASTA data shortly. 

Omar, can you post an example input file, please? Or tell me which one you'd been testing on?